intg/src/main/java/org/apache/atlas/type/AtlasEntityType.java

```diff
@@ -182,8 +182,11 @@ void resolveReferencesPhase3(AtlasTypeRegistry typeRegistry) throws AtlasBaseExc
 
             // validate if RelationshipDefs is defined for all entityDefs
             if (attributeEntityType != null && !hasRelationshipAttribute(attributeName)) {
-                LOG.warn("No RelationshipDef defined between {} and {} on attribute: {}.{}", getTypeName(),
-                         attributeEntityType.getTypeName(), getTypeName(), attributeName);
+                if(!typeRegistry.containsMissingRelationshipDefEntry(getTypeName(), attributeEntityType.getTypeName(), attributeName)) {
```

Comment:
consider replacing methods, containsMissingRelationshipDefEntry() and updateMissingRelationshipDefs(), with a single method like:

class AtlasTypeRegistry {
  void reportMissingRelationshipDef(String type1Name, String type2Name, String attrName) {
    if (!already-reported (type1Name,type2Name, attrName)) {
      LOG.warn("No RelationshipDef defined between {} and {} on attribute: {}.{}", type1Name, type2Name, attrName);

      add to already-reported list (type1Name, type2Name, attrName)
    }
  }
}